### PR TITLE
Update to kubernetes-1.15.4

### PR DESCRIPTION
**Pull request by sslavic**
Towards https://github.com/giantswarm/giantswarm/issues/6302

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -43,7 +47,7 @@ required = [
   name = "github.com/giantswarm/backoff"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/helmclient/pull/119 is merged.

File: pkg/framework/deployment/deployment.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func New(config Config) (*Deployment, error) {
 
 // Check ensures that key properties of the deployment are correct.
 func (d *Deployment) Check(name string, replicas int, expectedLabels, expectedMatchLabels map[string]string) error {
-	ds, err := d.k8sClient.Apps().Deployments(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	ds, err := d.k8sClient.AppsV1().Deployments(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
```
Comment (sslavic) [inline pkg/framework/deployment/deployment.go]: Deprecated versionless API was removed via https://github.com/kubernetes/client-go/commit/d2861e956fe1b2303c7a1688339f4aa2136c8c76#diff-76b263f1379ed2a72308d69111a53315

File: Gopkg.toml [hunk 3]
```diff
@@ -98,6 +102,34 @@ required = [
   branch = "master"
   name = "golang.org/x/oauth2"
 
+[[override]]
+  name = "github.com/giantswarm/k8sportforward"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be updated to "master" once https://github.com/giantswarm/k8sportforward/pull/19 gets merged.

**Review by TheoBrigitte — APPROVED** [review]
LGTM when tested

File: Gopkg.toml [hunk 4]
```diff
@@ -35,15 +35,15 @@ required = [
   branch = "master"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: This is to be reverted once https://github.com/giantswarm/apprclient/pull/26 is merged.

**Pull request merged**